trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/webapp/SamplingSerializer.java

```diff
@@ -43,17 +47,14 @@ public SamplingSerializer() {
 
     @Override
     public void serialize(ISampling value, JsonGenerator gen, SerializerProvider provider) throws IOException {
+        // Note that ISampling is not serialized as a json object, but only as json field.
+        // Hence it has to be embedded in an outer JSON object
         if (value instanceof Timestamps timestamps) {
-            gen.writeArray(timestamps.timestamps(), 0, timestamps.timestamps().length);
-
+            gen.writeObjectField("xValues", timestamps.timestamps()); //$NON-NLS-1$
         } else if (value instanceof Categories categories) {
-            gen.writeStartArray();
-            for (String category : categories.categories()) {
-                gen.writeString(category);
-            }
-            gen.writeEndArray();
-
+            gen.writeObjectField("xCategories", categories.categories()); //$NON-NLS-1$
         } else if (value instanceof Ranges timeRanges) {
+            gen.writeFieldName("xRanges"); //$NON-NLS-1$
```

Comment:
I believe it's not normal that an object serializer writes its own field name? I think it would be more correct if the field name was written in SeriesModelSerializer, but it would require another series of instanceof checks...
Or maybe if the Javadoc would make it clear that this is a special case and the serializer serializes it as a field. Could call it SamplingFieldSerializer.
Food for thought, optional change.